fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/digitalcash/DigitalCashHomeFragment.java

```diff
@@ -4,33 +4,86 @@
  * A simple {@link Fragment} subclass. Use the {@link DigitalCashHomeFragment#newInstance} factory
  * method to create an instance of this fragment.
  */
 public class DigitalCashHomeFragment extends Fragment {
+    private DigitalCashHomeFragmentBinding mBinding;
+    private DigitalCashViewModel mViewModel;
+
+    public DigitalCashHomeFragment() {
+        // not implemented yet
+    }
+
+    /**
+     * Use this factory method to create a new instance of this fragment using the provided
+     * parameters.
+     *
+     * @return A new instance of fragment DigitalCashHomeFragment.
+     */
+    public static DigitalCashHomeFragment newInstance() {
+        return new DigitalCashHomeFragment();
+    }
+
+    @Override
+    public View onCreateView(
+            @NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        this.mViewModel = DigitalCashActivity.obtainViewModel(getActivity());
+        mBinding = DigitalCashHomeFragmentBinding.inflate(inflater, container, false);
+        return mBinding.getRoot();
+    }
 
-  public DigitalCashHomeFragment() {
-    // not implemented yet
-  }
-  /**
-   * Use this factory method to create a new instance of this fragment using the provided
-   * parameters.
-   *
-   * @return A new instance of fragment DigitalCashHomeFragment.
-   */
-  public static DigitalCashHomeFragment newInstance() {
-    return new DigitalCashHomeFragment();
-  }
-
-  @Override
-  public View onCreateView(
-      LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
-    return inflater.inflate(R.layout.digital_cash_home_fragment, container, false);
-  }
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        setHomeInterface();
+    }
+
+    public void setHomeInterface() {
+        Lao lao = mViewModel.getCurrentLao();
+        if (lao == null) {
+            Toast.makeText(
+                            requireContext(),
+                            getString(R.string.digital_cash_please_enter_a_lao),
+                            Toast.LENGTH_SHORT)
+                    .show();
+        } else {
+            try {
+                PoPToken token = mViewModel.getKeyManager().getValidPoPToken(lao);
+                PublicKey publicKey = token.getPublicKey();
+                mBinding.digitalCashHomeAddress.setText(publicKey.getEncoded());
+                if (lao.getTransactionByUser().containsKey(publicKey)) {
+                    List<TransactionObject> transactions = lao.getTransactionByUser().get(publicKey);
+                    long totalAmount =
+                            TransactionObject.getMiniLaoPerReceiverSetTransaction(transactions, publicKey);
+                    mBinding.digitalCashSendAddress.setText(String.format("LAO coin : %s", totalAmount));
+                }
+
+            } catch (KeyException e) {
+                Toast.makeText(
```

Comment:
An improvement would be to use the `logAndShow` function of the `ErrorUtils` class. I think it's optional this close to the final demo